crates/oci-distribution/src/client.rs

```diff
@@ -438,126 +429,149 @@ mod test {
         )
     }
 
-    #[test]
-    fn test_to_v2_manifest() {
-        let c = Client::default();
-
-        for &(image, expected_uri) in [
-            (HELLO_IMAGE_NO_TAG, "https://webassembly.azurecr.io/v2/hello-wasm/manifests/latest"), // TODO: confirm this is the right translation when no tag
-            (HELLO_IMAGE_TAG, "https://webassembly.azurecr.io/v2/hello-wasm/manifests/v1"),
-            (HELLO_IMAGE_DIGEST, "https://webassembly.azurecr.io/v2/hello-wasm/manifests/sha256:51d9b231d5129e3ffc267c9d455c49d789bf3167b611a07ab6e4b3304c96b0e7"),
-            (HELLO_IMAGE_TAG_AND_DIGEST, "https://webassembly.azurecr.io/v2/hello-wasm/manifests/sha256:51d9b231d5129e3ffc267c9d455c49d789bf3167b611a07ab6e4b3304c96b0e7"),
-            ].iter() {
-                let reference = Reference::try_from(image).expect("failed to parse reference");
-                assert_eq!(c.to_v2_manifest_url(&reference), expected_uri);
-        }
+    #[rstest(
+        image, expected_uri,
+        case::no_tag(
+            HELLO_IMAGE_NO_TAG,
+            // TODO: confirm this is the right translation when no tag
```

Comment:
As far as I understand, no tag means it resolves to latest. We can update this later if it turns out to not be the case